src/dmd/dmodule.d

```diff
@@ -1355,3 +1382,221 @@ struct ModuleDeclaration
         return buf.extractString();
     }
 }
+
+struct ModuleComponentRange
+{
+    Identifiers* packages;
+    Identifier name;
+    size_t index;
+    @property auto totalLength() const { return packages.dim + 1; }
+
+    @property auto empty() { return index > packages.dim; }
+    @property auto front() const { return (index == packages.dim) ? name : (*packages)[index]; }
+    void popFront() { index++; }
+}
+
+// returns whether the given module should be included in the compilation
+private bool includeImportedModuleCheck(ModuleComponentRange components)
+    in { assert(global.params.includeImports); } body
```

Comment:
`in` would be flush with the declaration. Also, it looks like too much trouble - just plop the `assert` as the first line?